src/library_webgl.js

```diff
@@ -1479,7 +1483,7 @@ var LibraryGL = {
       if (GLctx.currentPixelUnpackBufferBinding) {
         GLctx['compressedTexSubImage2D'](target, level, xoffset, yoffset, width, height, format, imageSize, data);
       } else {
-        GLctx['compressedTexSubImage2D'](target, level, xoffset, yoffset, width, height, format, HEAPU8, data, imageSize);
+        if (imageSize) GLctx['compressedTexSubImage2D'](target, level, xoffset, yoffset, width, height, format, HEAPU8, data, imageSize);
```

Comment:
There's still the issue that no GL error is produced if the arguments are invalid. We could fix this with:

```
#if FULL_ES2
        else GLctx['compressedTexSubImage2D'](target, level, xoffset, yoffset, width, height, format, new Uint8Array);
#endif
```

and similar on all the other calls I changed in this PR.